test/controllers/fhir/r4/api_controller_test.rb

```diff
@@ -1036,6 +1036,67 @@ def setup_patients
     assert_equal DateTime.now.to_date, p.closed_at&.to_date
   end
 
+  test 'should differentiate USA and Foreign addresses in update' do
```

Comment:
I would also add a test for when a USA address is supplied with the USA extension as this test tests the USA address when it doesn't provide the extension.

Reply:
added in https://github.com/SaraAlert/SaraAlert/pull/686/commits/4abfeaa83b05bf05ab6ae8016c655a76af1643e0.